autoswitch_virtualenv.plugin.zsh

```diff
@@ -277,6 +277,12 @@ function _missing_error_message() {
     printf "\n"
 }
 
+function randstr()
+{
+    local strlen="${1:=4}"
+    python -c "import string, random; print(''.join(random.choice(string.ascii_lowercase) for _ in range($strlen)))"
```

Comment:
problem with this is that it only supports python3.

While we can assume most people would be using python3, I think there are still some cases where people want to work with legacy systems that use python2.

This might be less of an issue if `python3` is specifically used as the binary name?

Reply:
This is valid Python 2 as well!

Reply:
```
λ python2

WARNING: Python 2.7 is not recommended. 
This version is included in macOS for compatibility with legacy software. 
Future versions of macOS will not include Python 2.7. 
Instead, it is recommended that you transition to using 'python3' from within Terminal.

Python 2.7.16 (default, May  8 2021, 11:48:02) 
[GCC Apple LLVM 12.0.5 (clang-1205.0.19.59.6) [+internal-os, ptrauth-isa=deploy on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import string, random; print(''.join(random.choice(string.ascii_lowercase) for _ in range(4)))
vxbj
```

Reply:
Oh I saw your edit... Python 2 parses this as `print (parenthesised expression)`, not as a function call:
```
>>> sys.version_info
sys.version_info(major=2, minor=7, micro=16, releaselevel='final', serial=0)
>>> ast.dump(ast.parse('print(1)'))
'Module(body=[Print(dest=None, values=[Num(n=1)], nl=True)])'
>>> ast.dump(ast.parse('from __future__ import print_function; print(1)'))
"Module(body=[ImportFrom(module='__future__', names=[alias(name='print_function', asname=None)], level=0), Expr(value=Call(func=Name(id='print', ctx=Load()), args=[Num(n=1)], keywords=[], starargs=None, kwargs=None))])"

```
The problem is much bigger the other way round, i.e. print statements can't be made to work in py3.

Reply:
This works in both python 2 and 3:

```
 ~  python2 -c "from __future__ import print_function; print('hello')"                                     
hello
 ~  python3 -c "from __future__ import print_function; print('hello')"                                       
hello
```

so lets just add the `__future__` import and stop worrying about it :)